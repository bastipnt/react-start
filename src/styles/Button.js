import styled from 'styled-components';

const Button = styled.button`
  color: blue;
  border: 2px solid red;
`;

export default Button;
