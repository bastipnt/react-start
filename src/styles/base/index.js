export * from './Button';
export * from './Input';
export * from './List';
export * from './P';
export * from './Layout';
export * from './MD';
