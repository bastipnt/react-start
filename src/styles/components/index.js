export * from './ToDoList.styles';
export * from './Header.styles';
