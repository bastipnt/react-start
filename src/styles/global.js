import { injectGlobal } from 'styled-components';

/* eslint no-unused-expressions: 0 */
injectGlobal`
  html {
    box-sizing: border-box;
    -webkit-font-smoothing: antialiased;
    -webkit-tap-highlight-color: transparent;
    text-rendering: optimizeLegibility;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  html,
  body {
    height: 100%;
    width: 100%;
  }

  body {
    font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
    color: #fff;
  }

  body.fontLoaded {
    font-family: 'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  }

  #app {
    min-height: 100%;
    min-width: 100%;
    display: flex;
    flex-flow: row;
  }
`;
