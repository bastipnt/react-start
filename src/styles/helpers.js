// @flow
import CF from 'colors.js';

const validateHEX = (hex: string): string => {
  let hexWithoutHash = hex.match(/#(([a-f0-9]{3}){1,2})\b/i);
  if (!hexWithoutHash) hexWithoutHash = '000';
  else [, hexWithoutHash] = hexWithoutHash;
  const fullHex = hexWithoutHash.length < 6 ?
    hexWithoutHash[0] + hexWithoutHash[0] + hexWithoutHash[1] + hexWithoutHash[1] +
    hexWithoutHash[2] + hexWithoutHash[2] : hexWithoutHash;
  return `#${fullHex}`;
};

export const opacify = (hex: string = '#fff', opacity?: number = 1): string => {
  const validHex = validateHEX(hex);
  return `rgba(${CF.hex2rgb(validHex).a.toString()}, ${String(opacity)})`;
};
