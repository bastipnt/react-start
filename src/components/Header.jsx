import React from 'react';
import { Route, Link, Switch } from 'react-router-dom';
import { HeaderBar, HeaderTitle, HeaderLink } from 'styles';
import routes from '../routes';

export const DefaultHeader = () => (
  <HeaderBar>
    <HeaderTitle>Welcome to React Start</HeaderTitle>
    <HeaderLink><Link to="/todo" href="/todo" label="ToDo">ToDo Example</Link></HeaderLink>
  </HeaderBar>
);

const Header = () => (
  <Switch>
    {routes.map(({
      exact, key, header = DefaultHeader, path,
    }: ROUTE) => (
      <Route
        exact={exact}
        key={key}
        path={path}
        component={header}
      />
    ))}
  </Switch>
);

export default Header;
