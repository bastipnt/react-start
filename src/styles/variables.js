// @flow
import { opacify } from './helpers';

export const color = {
  text: opacify('#70708D', 1),
  textLight: opacify('#70708D', 0.4),
  textSpecial: '#6064FA',
  textHeader: opacify('#fff', 0.8),
  background: opacify('#fff', 1),
  backgroundDark: opacify('#70708D', 0.1),
  backgroundSpecial: '#FDFDFF',
  backgroundButton: opacify('#fff', 0.5),
  divider: opacify('#ECF2F6', 1),
  dividerLight: opacify('#ECF2F6', 1),
  placeholder: opacify('#6064FA', 0.5),
  checkboxBorder: opacify('#ECF2F6', 1),
  checkboxBackground: opacify('#6064FA', 1),
  checkboxText: opacify('#fff', 1),
  button: opacify('#70708D', 1),
  header: opacify('#fff', 0.1),
};
