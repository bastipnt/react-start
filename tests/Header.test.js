import React from 'react';
import { shallow } from 'enzyme';
import { HeaderTitle } from 'styles';
import { Link, Route } from 'react-router-dom';

import Header, { DefaultHeader } from '../src/components/Header';

let component;

describe('<Header />', () => {
  beforeAll(() => { component = shallow(<Header />); });

  it('should render the header', () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it('should render some routes', () => {
    expect(component.find(Route).length).not.toBe(0);
  });
});

describe('default <Header />', () => {
  beforeAll(() => { component = shallow(<DefaultHeader />); });

  it('should render the default header', () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it('should render a title', () => {
    expect(component.find(HeaderTitle).length).toBe(1);
  });

  it('should render a link to the toDo example', () => {
    expect(component.find(Link).length).toBe(1);
    expect(component.find(Link).prop('to')).toEqual('/todo');
  });
});
