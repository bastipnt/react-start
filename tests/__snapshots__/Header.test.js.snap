// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header 1`] = `
Array [
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={undefined}
      path="/todo"
    />
  </Switch>,
]
`;

exports[`default <Header /> should render the default header 1`] = `
Array [
  <Headerstyles__HeaderBar>
    <Headerstyles__HeaderTitle>
      Welcome to React Start
    </Headerstyles__HeaderTitle>
    <Headerstyles__HeaderLink>
      <Link
        href="/todo"
        label="ToDo"
        replace={false}
        to="/todo"
      >
        ToDo Example
      </Link>
    </Headerstyles__HeaderLink>
  </Headerstyles__HeaderBar>,
]
`;
